Modules/Core/Common/include/itkFixedArray.h

```diff
@@ -163,33 +163,30 @@ class ITK_TEMPLATE_EXPORT FixedArray
   template< typename TFixedArrayValueType >
   FixedArray(const FixedArray< TFixedArrayValueType, VLength > & r)
   {
-    typename FixedArray< TFixedArrayValueType, VLength >::ConstIterator input = r.Begin();
-    Iterator i = this->Begin();
-    while ( i != this->End() )
-      {
-      *i++ = static_cast< TValue >( *input++ );
-      }
+    auto input = r.cbegin();
+
+    for (auto& element : m_InternalArray)
+    {
+      element = static_cast< TValue >( *input++ );
+    }
   }
 
   template< typename TScalarValue >
   FixedArray(const TScalarValue *r)
-    {
-      std::copy(r, r + this->Size(), this->GetDataPointer());
-    }
+  {
+    std::copy_n(r, VLength, m_InternalArray);
+  }
 
   /** Operator= defined for a variety of types. */
   template< typename TFixedArrayValueType >
   FixedArray & operator=(const FixedArray< TFixedArrayValueType, VLength > & r)
   {
-    if ( (const void *)r.Begin() != (const void *)m_InternalArray )
```

Comment:
The equality check has been removed. If it was intentional, what is the reasoning behind it?

Reply:
@dzenanz Thanks for your remark, I appreciate that you review so carefully!

Yes indeed, the check was intentionally removed. How could `this` and `r` possibly have their internal arrays starting at the very same memory address? Note that their value types are different, already, for this `operator=` overload. An assignment that just copies fixed arrays of the same type calls the default `operator=(const FixedArray &)`, instead.